options/internal/x86/setjmp.S

```diff
@@ -17,8 +17,8 @@ __setjmp:
 	xor %eax, %eax
 	ret
 
-  1:
-	jmp __sigsetjmp
+1:
+	jmp __sigsetjmp@PLT
```

Comment:
Why do we need `@PLT` for a jump within the same DSO?

Reply:
The PLT jump is needed to avoid gcc emitting relocations in (read-only) `.text.

Reply:
Is that allowed?

Reply:
Relocations in `.text` set a flag that we don't handle yet, and would require us to map `.text` as writable for the time of performing relocations (which isn't really optimal IMO). This solution removes the necessity to handle this for now.

Reply:
Is this flag marked as TODO somewhere?

Reply:
The flag is `DF_TEXTREL`, and the TODO is here: https://github.com/managarm/mlibc/blob/47e9748fd2700c89ed3ec5a6908b44dbf3f51da7/options/rtdl/generic/linker.cpp#L627-L633